tgg/research/gnomad_tgps.py

```diff
@@ -0,0 +1,239 @@
+producing a pangenome of high quality long read sequencing of 350 diverse humans
+which will become the new pangenome reference. The PSR group is responsible to
+selecting samples to be included in the pangenome, which should comprehensively cover
+most common variants, defined as variants >1%, in human populations globally. We are
+halfway through (phase 1) of the project, and most of the sampling has been selected
+from the 1 000 Genome Cell lines (N=174 samples). We have developed a simple variant
+counting metric that helps us evaluate anticipated performance of the pangenome using
+short read data in phase one samples and additional out-of-sample genomes from the
+1KG project. Here is a link to a brief slide deck that describes this: [
+https://www.dropbox.com/s/fo5nrxerxwukw7x/20220908.HPRC-PopulationSampling.Gnomad.pdf
+?dl=0]
+
+We would love to additionally be able to assess the performance in gnomAD – not only
+is this an independently ascertained dataset, but it would also give us a look ahead
+to how aligned the pangenome will be to the top resource used by clinical testing
+labs. Ideally, we would like to run the analysis on individual-level data in gnomAD,
+and I’m reaching out to see if you would be willing to collaborate with us on this
+analysis. The script is a very simple variant count algorithm, so hopefully very
+straightforward to run. We are currently writing up this work, and of course,
+anybody who participates in this analysis would be included as a co-author."""
+
+import hail as hl
+from gnomad_qc.v3.resources.basics import get_gnomad_v3_vds
+from gnomad_qc.v3.resources.release import release_sites
+from gnomad_qc.v3.resources.meta import meta
+import hailtop.fs as hfs
+import argparse
+import logging
+
+NUM_OF_GNOMAD_SAMPLES = 76156
+OUTPUT_FILENAME = "pangemoe_assessment_stats.txt"
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("Pangenome_assessment")
+logger.setLevel(logging.INFO)
+
+
+def get_pangenome_ids(
+        id_path: str,
+) -> hl.expr.expressions.typed_expressions.SetExpression:
+    """
+    Generate a Hail set of sample IDs in the pangenome.
+
+    :param str id_path: Cloud path to read in the .txt file containing pangenome sample
+    IDs.
+    :return: A set of sample IDs as hl.expr.expressions.typed_expressions.SetExpression.
+    """
+    ids = []
+    with hfs.open(id_path, 'r') as file:
+        for line in file:
+            sid = line.split(' ')[1].strip()
+            ids.append(sid)
+    ids = hl.set(ids)
+    return ids
+
+
+def write_output(
+        dat: hl.utils.struct.Struct,
+        file_dir: str,
+        filename: str,
+) -> None:
+    """
+    Write the output Struct containing computed stats to a .txt file.
+
+    :param hl.utils.struct.Struct dat: Data containing computed stats.
+    :param str file_dir: Output directory.
```

Comment:
this also should be a bucket path, right? would be good to note that here, particularly for consistency (since the function above specifies that the input path is in the cloud)